File: Algorithms/FSharp/RandomSearch/Tests/RandomSearchTests.fs
namespace OSOL.Extremum.Algorithms.FSharp

open OSOL.Extremum.Cores.DotNet.Vectors
open OSOL.Extremum.Cores.DotNet.Optimization
open OSOL.Extremum.Cores.DotNet.Optimization.Testing

open Xunit

module RandomSearchTests = 

    let r: double = 1.0;
    let oneMin: double = 60.0;
    
    [<Fact>]
    let TestRandomSearch() =
        let tester = new RealTester()
        let configs = 
            [|
                RandomSearch.CreateFixedStepRandomSearch (1.0 * r) (1.0 * oneMin)
                RandomSearch.CreateFixedStepRandomSearch (0.5 * r) (2.0 * oneMin)
                RandomSearch.CreateFixedStepRandomSearch (0.1 * r) (5.0 * oneMin)
            |]
        Assert.True(tester.Check configs)

File: Algorithms/CSharp/RandomSearch/Tests/RandomSearchTests.cs
using OSOL.Extremum.Cores.DotNet.Vectors;
using OSOL.Extremum.Cores.DotNet.Optimization;
using OSOL.Extremum.Cores.DotNet.Optimization.Testing;
using Xunit;

namespace OSOL.Extremum.Algorithms.CSharp
{
    public static class RandomSearchTests
    {
        private static double r = 1.0;
        private static double oneMin = 60.0;

        [Fact]
        static void TestRandomSearch()
        {
            var tester = new RealTester();
            Assert.True(tester.Check(
                RandomSearch.CreateFixedStepRandomSearch(radius: (1.0 * r), maxTime: 1 * oneMin),
                RandomSearch.CreateFixedStepRandomSearch(radius: (0.5 * r), maxTime: 2 * oneMin),
                RandomSearch.CreateFixedStepRandomSearch(radius: (0.1 * r), maxTime: 5 * oneMin)));
        }
    }
}